presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -228,6 +238,19 @@ public long getValidationCpuNanos()
                 .mapToLong(HiveWriter::getValidationCpuNanos)
                 .sum();
 
+        if (waitForFileRenaming && verificationTasks.isEmpty()) {
+            ImmutableList.Builder<Slice> partitionUpdatesWithRenamedFileNames = ImmutableList.builder();
+            List<ListenableFuture<?>> futures = new ArrayList<>();
+            for (int i = 0; i < writers.size(); i++) {
+                ListenableFuture<?> fileNameFuture = toListenableFuture(hiveMetadataUpdater.getMetadataResult(i));
+                SettableFuture renamingFuture = SettableFuture.create();
+                futures.add(renamingFuture);
+                int writerIndex = i;
```

Comment:
this assignment looks unnecessary

Reply:
If i use `i` as input param to lambda function then i am getting syntax error `Variable used in lambda expression should be final or effectivelyfinal`. Hence added this line.